src/msix/pack/AppxPackageWriter.cpp

```diff
@@ -2,26 +2,60 @@
 //  Copyright (C) 2019 Microsoft.  All rights reserved.
 //  See LICENSE file in the project root for full license information.
 // 
+
+#include "AppxPackaging.hpp"
 #include "AppxPackageWriter.hpp"
 #include "MsixErrors.hpp"
 #include "Exceptions.hpp"
+#include "ContentType.hpp"
+#include "SHA256.hpp"
 
 #include <string>
+#include <memory>
+#include <future>
+#include <algorithm>
+#include <functional>
 
 namespace MSIX {
 
+    static const std::uint32_t defaultBlockSize = 65536;
+
     AppxPackageWriter::AppxPackageWriter(IStream* outputStream) : m_outputStream(outputStream)
     {
         m_state = WriterState::Open;
+        m_blockMapWriter = std::make_unique<BlockMapWriter>();
+        m_contentTypeWriter = std::make_unique<ContentTypeWriter>();
     }
 
     // IPackageWriter
     void AppxPackageWriter::Pack(const ComPtr<IDirectoryObject>& from)
     {
         ThrowErrorIf(Error::InvalidState, m_state != WriterState::Open, "Invalid package writer state");
         auto fileMap = from->GetFilesByLastModDate();
+        std::vector<std::unique_ptr<PayloadFile>> files;
+        for(const auto& file : fileMap)

```

Comment:
TODO: Special case footprint file types.

Reply:
Probably after other files so they don't mess up the default.  Unless that is why you put in the force override.

---
In reply to: [278241762](https://github.com/Microsoft/msix-packaging/pull/95#discussion_r278241762) [](ancestors = 278241762)

Reply:
Actually GetFilesByLastModDate shouldn't return any footprint file. The only footprint file we look in a given directory to pack is the AppxManifest. Any other should be just ignore them. I'll rename it to GetPayloadFilesByLastModDate, we could us the StorageObject FileNameOptions enum, but I don't see any other use case that require this map with payload files

And yes, the force override is going to be used for the footprint files, specially with the AppxSignature.p7x

---
In reply to: [278242004](https://github.com/Microsoft/msix-packaging/pull/95#discussion_r278242004) [](ancestors = 278242004,278241762)